build/Targets/Imports.targets

```diff
@@ -404,25 +407,21 @@
     </PropertyGroup>
   </Target>
 
+  <!-- Guard against MSBuild picking the wrong SDK version. Its resolver does not always properly enforce
+       the SDK specified in global.json 
+       
+       https://github.com/dotnet/core-setup/issues/3805
+       -->
   <Target Name="BeforeBuild" Condition="'$(MSBuildRuntimeType)' != 'Core'">
     <ValidateBuildEnvironment
       MSBuildBinPath="$(MSBuildBinPath)"
       MSBuildMinimumFileVersion="15.3.409"
       MSBuildMinimumDisplayVersion="MSBuild 15.3"
       />
 
-    <PropertyGroup>
-      <ShortSdkVersion Condition=" $(dotnetSdkVersion.IndexOf('-')) > 0 ">$(dotnetSdkVersion.Substring(0, $(dotnetSdkVersion.IndexOf('-'))))</ShortSdkVersion>
-      <ShortSdkVersion Condition=" $(dotnetSdkVersion.IndexOf('-')) &lt;= 0 ">$(dotnetSdkVersion)</ShortSdkVersion>
-    </PropertyGroup>
-
-    <!-- The $(UsingMicrosoftNETSdk) property is set when we are using a 2.0 SDK or newer (non-legacy). This means
-         we can depend on global.json having enforced the minimum version we require for our build. If it's not
-         set though we need to error out and provide a link to get the correct SDK installed.
-      -->
     <Error 
-      Text="The $(ShortSdkVersion) SDK is required to build this repo. It can be install here https://dotnetcli.blob.core.windows.net/dotnet/Sdk/2.2.0-preview1-007622/dotnet-sdk-2.2.0-preview1-007622-win-x64.exe"
-      Condition="'$(UsingMicrosoftNETSdk)' == ''" />
+      Text="The $(dotnetSdkVersion) SDK is required to build this repo but found $(NETCoreSdkVersion) instead. The correct SDK can be install here https://dotnetcli.blob.core.windows.net/dotnet/Sdk/2.2.0-preview1-007622/dotnet-sdk-2.2.0-preview1-007622-win-x64.exe"
```

Comment:
note that this will say found [blank] instead for anything earlier than 2.1.300-preview2. You may want to condition the message.